gax-java/gax/src/test/java/com/google/api/gax/util/TimeConversionTestUtils.java

```diff
@@ -0,0 +1,118 @@
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.util.function.Function;
+
+public class TimeConversionTestUtils {
+
+  public static <Target> void testInstantMethod(
+      Long testValue,
+      Function<java.time.Instant, Target> javaTimeTargetSupplier,
+      Function<org.threeten.bp.Instant, Target> threetenTargetSupplier,
+      Function<Target, java.time.Instant> javaTimeGetter,
+      Function<Target, org.threeten.bp.Instant> threetenGetter) {
+    Function<java.time.Instant, Long> javaTimeTester = value -> value.toEpochMilli();
+    Function<org.threeten.bp.Instant, Long> threetenTester = value -> value.toEpochMilli();
+    java.time.Instant javaTimeSupplierValue =
+        testValue == null ? null : java.time.Instant.ofEpochMilli(testValue);
+    org.threeten.bp.Instant threetenSupplierValue =
+        testValue == null ? null : org.threeten.bp.Instant.ofEpochMilli(testValue);
+    testTimeObjectMethod(
+        testValue,
+        javaTimeSupplierValue,
+        javaTimeTargetSupplier,
+        javaTimeGetter,
+        threetenGetter,
+        javaTimeTester,
+        threetenTester);
+    testTimeObjectMethod(
+        testValue,
+        threetenSupplierValue,
+        threetenTargetSupplier,
+        javaTimeGetter,
+        threetenGetter,
+        javaTimeTester,
+        threetenTester);
+  }
+
+  public static <Target> void testDurationMethod(
+      Long testValue,
+      Function<java.time.Duration, Target> javaTimeTargetSupplier,
+      Function<org.threeten.bp.Duration, Target> threetenTargetSupplier,
+      Function<Target, java.time.Duration> javaTimeGetter,
+      Function<Target, org.threeten.bp.Duration> threetenGetter) {
```

Comment:
Can you add some quick/ small comments or javadocs to explain what this helper method is doing? Also for the two other helper methods in here?